mgmtsystem_nonconformity/models/mgmtsystem_nonconformity_cause.py

```diff
@@ -0,0 +1,71 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+#
+#    OpenERP, Open Source Management Solution
+#    Copyright (C) 2010 Savoir-faire Linux (<http://www.savoirfairelinux.com>).
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import models, fields, api
+from openerp.osv import osv
+
+
+class mgmtsystem_nonconformity_cause(models.Model):
+    """
+    Cause of the nonconformity of the management system
+    """
+    _name = "mgmtsystem.nonconformity.cause"
+    _description = "Cause of the nonconformity of the management system"
+    _order = 'parent_id, sequence'
```

Comment:
It finds by itself how to construct the left and right? I understand that it creates a structure that will allow fetching part of the trees efficiently but the doc doesn't tell much how it finds the left and right. I'll add it and lets see how it goes.